arangod/Aql/PruneExpressionEvaluator.cpp

```diff
@@ -45,6 +45,5 @@ bool PruneExpressionEvaluator::evaluate() {
   bool mustDestroy = false;
   aql::AqlValue res = _pruneExpression->execute(&_ctx, mustDestroy);
   arangodb::aql::AqlValueGuard guard(res, mustDestroy);
-  TRI_ASSERT(res.isBoolean());
```

Comment:
Yes, we can actually cast many thing to bool, so not required to assert that we have a bool in hand